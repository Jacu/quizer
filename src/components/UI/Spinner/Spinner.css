.Spinner {
    display: flex;
    width: 64px;
    height: 64px;
    margin: 36px auto;
  }
  .Spinner:after {
    content: " ";
    display: block;
    width: 46px;
    height: 46px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid #125271;
    border-color: #125271 transparent #125271 transparent;
    animation: lds-dual-ring 1.2s linear infinite;    
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
  