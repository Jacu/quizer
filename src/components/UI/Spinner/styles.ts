import styled, { keyframes } from "styled-components";
import { PRIMARY_THEME } from "~/utils/colors";

const lds_dual_ring = keyframes`
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
`;

export const Spinner = styled.div`
  display: flex;
  width: 64px;
  height: 64px;
  margin: 36px auto;
  
  :after {
    content: " ";
    display: block;
    width: 46px;
    height: 46px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid #125271;
    border-color: ${PRIMARY_THEME.BASE} transparent ${PRIMARY_THEME.BASE} transparent;
    animation: ${lds_dual_ring} 1.2s linear infinite;
  }
`;