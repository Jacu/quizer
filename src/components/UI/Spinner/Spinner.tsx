import React from 'react';
import * as styled from './styles'

const Spinner = () => <styled.Spinner/>

export default Spinner;