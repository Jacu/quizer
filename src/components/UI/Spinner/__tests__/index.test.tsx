
import React from 'react';
import Spinner from '../';
import toJson from "enzyme-to-json";
import {mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Spinner', () => {
    it('should render', () => {
        const wrapper = mount(<Spinner />);
        expect(toJson(wrapper,{ mode: 'deep' })).toMatchSnapshot();
    });
});