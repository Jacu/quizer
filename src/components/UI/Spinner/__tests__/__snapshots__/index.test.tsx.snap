// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render 1`] = `
<styled.div>
  <div
    className="sc-AxjAm iSIMQH"
  />
</styled.div>
`;
